src/libraries/System.Runtime/tests/System/Reflection/ReflectionCacheTests.cs

```diff
@@ -47,6 +57,10 @@ public void GetMembers_MultipleCalls_SameObjects()
             AssertSameEqualAndHashCodeEqual(fi1, fi2);
             AssertSameEqualAndHashCodeEqual(ei1, ei2);
             AssertSameEqualAndHashCodeEqual(ci1, ci2);
+
+            PropertyInfo parentProperty = typeof(A).GetProperty("P");
+            PropertyInfo childProperty = s_type.GetProperty("P");
+            Assert.NotEqual(parentProperty, childProperty);
```

Comment:
We should add `Assert.NotNull(parentProperty )` and  `Assert.NotNull(childProperty)` here and the other cases below to make sure `null` isn't being return from `GetProperty()`